### DISCUSSION TITLE
Feature extraction and frame length dependency

### DISCUSSION THREAD

**Discussion by jonaslindstrm** [blue-yonder/tsfresh#1054] [Q&A] [answered]
Hi, first of all, great job with this library. I use it quite alot!

But i got a some questions regarding regarding the feature calculations and frame length dependencies. 

To give some background about my question:
I input a list of dataframes into tsfresh and then i have a classification problem where one class tends
to have shorter length of the frames than the other. 

When i input a random column to each of theese frames i can see that the random features get quite high importance to the prediction. 
Hence the question about the length dependency. 

I use the EfficientFCParameters settings where i filter out the parameters which i suspect the most:
  settings = EfficientFCParameters()
  del settings['length']
  del settings['sum_values']
  del settings['sum_of_reoccurring_values']
  del settings['sum_of_reoccurring_data_points']
  del settings['range_count']
  del settings['abs_energy'] 

Of course it could be something in my dataset why this happens, i would just like to know there exist some settings which are not normalized by the length of the frame?

Thanks!
Br 
Jonas

**Comment by nils-braun** [comment, marked as answer]
Hi @jonaslindstrm - sorry for the late (and also not very useful answer). As far as I know there has not been any study on the dependency of the length on the feature extractors, so I do not know which of them would have the least dependency. 
One way to test this would be exactly like you did: testing with some random data/artificial data. Another possibility is to normalize your data to have the same/similar length - either by padding or even better by re-sampling or windowing.

**Reply by jonaslindstrm** [reply]
Hi @nils-braun, thanks for the reply! I think you answered my question and also given me some useful advice. 
Sometimes the answer is not as straightfoward as one would like, thats just life :)

Br 
Jonas
